docs-website/src/theme/Root.js

```diff
@@ -0,0 +1,22 @@
+import React, { useEffect } from 'react';
+import { useLocation } from '@docusaurus/router';
+
+export default function Root({ children }) {
+  const location = useLocation();
+
+  useEffect(() => {
+    if (location.hash) {
+        console.log("Hash found:", location.hash);
+        const id = location.hash.replace('#', '');
+        const el = document.getElementById(id);
+        console.log("Element:", el);
+      if (el) {
+        setTimeout(() => {
+          el.scrollIntoView({ behavior: 'auto' });
+        }, 0);
```

Comment:
what's this for? I thought browsers are supposed to do this automatically?

e.g. the hash locations work fine on all other pages?

Reply:
this is a hacky solution to handle external anchor links not landing on the intended section. somehow this issue still persists (lands at top when navigating from outside), so keeping this file until we have a better fix